airflow/providers/amazon/aws/hooks/athena.py

```diff
@@ -133,7 +133,17 @@ def get_query_results(self, query_execution_id):
         elif query_state in self.INTERMEDIATE_STATES or query_state in self.FAILURE_STATES:
             self.log.error('Query is in {state} state. Cannot fetch results'.format(state=query_state))
             return None
-        return self.get_conn().get_query_results(QueryExecutionId=query_execution_id)
+        result_params = {'QueryExecutionId': query_execution_id,
+                         'MaxResults': max_results}
+        results = self.get_conn().get_query_results(**result_params)
```

Comment:
Does boto have a built-in pager for this?

Reply:
Yes, see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/athena.html#paginators and https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html

Reply:
Yes that does make sense.  Now that I think about it, it probably makes sense to just add a new function to check that the query is successful and then return the paginator. For the get_query_results function maybe we should just adding the option to send in the NextTokenId so you can paginate manually if you want.